modules/schema/src/main/scala/lucuma/odb/json/sequence.scala

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.json
+
+import cats.syntax.either.*
+import io.circe.Decoder
+import io.circe.DecodingFailure
+import io.circe.Encoder
+import io.circe.Json
+import io.circe.syntax.*
+import lucuma.core.enums.Breakpoint
+import lucuma.core.enums.Instrument
+import lucuma.core.math.Offset
+import lucuma.core.math.Wavelength
+import lucuma.core.model.sequence.Atom
+import lucuma.core.model.sequence.DynamicConfig.GmosNorth
+import lucuma.core.model.sequence.DynamicConfig.GmosSouth
+import lucuma.core.model.sequence.ExecutionSequence
+import lucuma.core.model.sequence.FutureExecutionConfig
+import lucuma.core.model.sequence.StaticConfig
+import lucuma.core.model.sequence.Step
+import lucuma.core.model.sequence.StepConfig
+import lucuma.core.model.sequence.StepTime
+import lucuma.core.util.TimeSpan
+import lucuma.core.util.Uid
+
+trait SequenceCodec {
+
+  import gmos.given
+  import stepconfig.given
+  import time.decoder.given
+
+  given Decoder[StepTime] =
+    Decoder.instance { c =>
+      for {
+        h <- c.downField("configChange").as[TimeSpan]
+        e <- c.downField("exposure").as[TimeSpan]
+        r <- c.downField("readout").as[TimeSpan]
+        w <- c.downField("write").as[TimeSpan]
+        t <- c.downField("total").as[TimeSpan]
+      } yield StepTime(h, e, r, w, t)
+    }
+
+  given (using Encoder[TimeSpan]): Encoder[StepTime] =
+    Encoder.instance { (a: StepTime) =>
+      Json.obj(
+        "configChange" -> a.configChange.asJson,
+        "exposure"     -> a.exposure.asJson,
+        "readout"      -> a.readout.asJson,
+        "write"        -> a.write.asJson,
+        "total"        -> a.total.asJson
+      )
+    }
+
+  given given_Decoder_Step_GmosNorth: Decoder[Step.GmosNorth] =
```

Comment:
Are you using these instances by name anywhere?

Reply:
No, but the automatically generated names clash.